crates/swc_ecma_codegen/src/lib.rs

```diff
@@ -718,17 +718,25 @@ where
         srcmap!(self, num, true);
 
         if self.cfg.minify {
-            value = minify_number(num.value);
-            self.wr.write_str_lit(DUMMY_SP, &value)?;
+            if num.value.is_infinite() && num.raw.is_some() {
+                self.wr.write_str_lit(DUMMY_SP, &num.raw.clone().unwrap())?;
+            } else {
+                value = minify_number(num.value);
+                self.wr.write_str_lit(DUMMY_SP, &value)?;
+            }
         } else {
             match &num.raw {
                 Some(raw) => {
                     if raw.len() > 2 && self.cfg.target < EsVersion::Es2015 && {
                         let slice = &raw.as_bytes()[..2];
                         slice == b"0b" || slice == b"0o" || slice == b"0B" || slice == b"0O"
                     } {
-                        value = num.value.to_string();
-                        self.wr.write_str_lit(DUMMY_SP, &value)?;
+                        if num.value.is_infinite() && num.raw.is_some() {
```

Comment:
Isn't this logic duplicate of the logic above?

Reply:
the condition is the same but the statement isn't